snap/info.go

```diff
@@ -892,12 +892,13 @@ func (e AttributeNotFoundError) Is(target error) bool {
 type PlugInfo struct {
 	Snap *Info
 
-	Name      string
-	Interface string
-	Attrs     map[string]interface{}
-	Label     string
-	Apps      map[string]*AppInfo
-	Hooks     map[string]*HookInfo
+	Name           string
+	Interface      string
+	Attrs          map[string]interface{}
+	Label          string
+	Apps           map[string]*AppInfo
+	Hooks          map[string]*HookInfo
+	ComponentHooks map[string]*HookInfo
```

Comment:
ComponentHooks must be separate from Hooks because they might share the same name, which is the key in the maps.

Reply:
what's the key of this? given that the many components can have the same hooks but they can be bound to different plugs for each component

Reply:
The key was the hook name. But you're right, this data structure would only let a plug point to a single component hook of a specific name.

>maybe we need a mapping -> -> hook-info

I think you dropped a word there? Maybe a mapping of some tuple of component name + hook name -> hook-info?